File: Inheritance.cpp
#include<bits/stdc++.h>

using namespace std;

class Vechicle{
    public:
    string name;
    int tire;
    

    void display_name(){
        cout<<"Car Name is : "<<name<<endl;
    }
};

class Car : public Vechicle{
    public:
    int tire = 4;

    Car(string name){
        this->name = name;
        
    }

    void feature(){
        cout<<"Extra feature"<<endl;
    }
};

int main(){
    Car porche("Porche");
    cout<<porche.tire<<endl;
    porche.feature();     
    porche.display_name();     
    return 0;
}

File: Inheritance.py
class Vechicle:
    def __init__(self,name) -> None:
        self.name = name
    def display_name(self):
        print("Vehicle name",self.name)

class Car(Vechicle):
    def __init__(self, name) -> None:
        Vechicle.__init__(self,name)
        self.tire = 4

porche = Car('porche')
print(porche.tire)
porche.display_name()